src/main/java/roomescape/dao/ReservationDAO.java

```diff
@@ -26,57 +27,66 @@ public ReservationDAO(JdbcTemplate jdbcTemplate) {
         this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(jdbcTemplate);
     }
 
-    public Reservation addReservation(final Reservation reservation) {
-        final String query = "INSERT INTO reservation (name, date, time) VALUES (:name, :date, :time)";
-        KeyHolder keyHolder = new GeneratedKeyHolder();
+    public Reservation add(final Reservation reservation) {
+        SimpleJdbcInsert jdbcInsert = new SimpleJdbcInsert(jdbcTemplate)
+                .withTableName("reservation")
+                .usingGeneratedKeyColumns("id");
 
-        MapSqlParameterSource params = new MapSqlParameterSource()
-                .addValue("name", reservation.getName())
-                .addValue("date", reservation.getDate().toString())
-                .addValue("time", reservation.getTime().toString());
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("name", reservation.getName());
+        parameters.put("date", reservation.getDate().toString());
+        parameters.put("time_id", reservation.getTime().getId());
```

Comment:
Map의 api를 통해 선언과 동시에 정의할 수도 있어요!
```suggestion
        Map<String, Object> parameters = Map.of(
                "name", reservation.getName(),
                "date", reservation.getDate().toString(),
                "time_id", reservation.getTime().getId()
        );
```